tiri/src/lib.rs

```diff
@@ -45,67 +45,47 @@ const DUMP_CONTEXT: usize = 4;
 /// The contents of each local variable is represented by a `Value`.
 #[derive(Debug, Clone, Eq, PartialEq)]
 enum Value {
-    /// A value that can be stored immediately into a local variable without touching memory.
-    Immediate(Immediate),
-    /// A reference to a potentially larger (and potentially composite) value in memory (either on
-    /// the heap or stack).
-    _Indirect, // FIXME Not implemented.
+    /// A small (u128-sized or smaller) value.
+    Scalar(Scalar),
+    /// A pair of the above, used for checked operations.
+    ScalarPair(Scalar, Scalar),
+    /// A reference to a potentially larger value anywhere in the interpreter's memory.
+    _Memory, // FIXME Not implemented.
 }
 
 impl Value {
     /// A helper to create a raw scalar value of the given value and size.
     #[cfg(test)]
     fn raw_scalar(data: u128, size: u8) -> Self {
-        Value::Immediate(Immediate::Scalar(Scalar::Raw { data, size }))
+        Self::Scalar(Scalar::Raw { data, size })
     }
 
     /// A helper to cast a value to a Boolean, if appropriate.
     fn as_bool(&self) -> bool {
         match self {
-            Value::Immediate(i) => i.as_bool(),
+            // In the case of a `ScalarPair` we assume we are casting to bool to inspect the result
```

Comment:
"Assume" is a bit vague. Is this always the case or ...?

Reply:
My current thinking is that we shouldn't generate SIR where you want to cast the first element of a pair to a bool. Let me fiddle the comment.

Reply:
OK.